src/quo2/components/wallet/transaction_progress/view.cljs

```diff
@@ -0,0 +1,298 @@
+    (or (= network-state :pending) (= network-state :sending))  :pending
+    (= network-state :error)                                    :error
+    :else                                                       :confirmed))
+
+(defn calculate-box-width
+  [showHalf]
+  (cond
+    (and showHalf (< @lcounter 30)) (- total-box @lcounter)
+    showHalf                        30
+    (< @lcounter total-box)         (- total-box @lcounter)
+    :else                           0))
+
+(defn progress-boxes-arbitrum
+  [network-state network-type]
+  [rn/view
+   {:style style/progress-box-container}
+   [progress-box/view
+    {:state               (calculate-box-state-arbitrum network-state network-type)
+     :customization-color :blue}]
+   [rn/view
+    {:style            style/progress-box-arbitrum
+     :background-color (colors/theme-colors colors/white colors/neutral-5 colors/neutral-70)
+     :border-color     (colors/theme-colors colors/white colors/neutral-10 colors/neutral-80)}
+    [rn/view
+     (assoc
+      (let [box-style (cond
+                        (= network-state :finalising) (assoc {:style style/progress-box-arbitrum-abs}
+                                                             :right (str (calculate-box-width true) "%")
+                                                             :background-color
+                                                             (colors/theme-colors
+                                                              (colors/custom-color :blue 50)
+                                                              (colors/custom-color :blue 60)
+                                                              quo.theme))
+                        (= network-state :finalized)  (assoc {:style style/progress-box-arbitrum-abs}
+                                                             :right (str (calculate-box-width false) "%")
+                                                             :background-color
+                                                             (colors/theme-colors
+                                                              (colors/custom-color :blue 50)
+                                                              (colors/custom-color :blue 60)
+                                                              quo.theme))
+                        :else                         (assoc {:style style/progress-box-arbitrum-abs}
+                                                             :background-color
+                                                             (colors/theme-colors colors/white
+                                                                                  colors/neutral-5
+                                                                                  colors/neutral-70)))]
+        box-style)
+      :align-self "flex-end"
+      :border-color
+      (colors/theme-colors colors/white colors/neutral-10 colors/neutral-80))]]])
+
+(defn render-text
```

Comment:
We try to name components as nouns to help everybody understand that Reagent components shouldn't be called as functions. So `render-text` could be `text-component`, `render-title` could `title-component`, and so on. When the name is too common (e.g. `text` or `title`) and you don't want to shadow your own bindings `title` or `text`, it's better to add some kind of suffix to the component names. Guideline: https://github.com/status-im/status-mobile/blob/79daa4d5290626ccc7e157f9600685722054e09e/doc/new-guidelines.md#use--instead-of--in-reagent-components